src/components/SearchList.tsx

```diff
@@ -0,0 +1,48 @@
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemText from "@mui/material/ListItemText";
+import ListItemAvatar from "@mui/material/ListItemAvatar";
+import Avatar from "@mui/material/Avatar";
+import { useNavigate } from "react-router-dom";
+import { useContext } from "react";
+import { AppContext } from "../App";
+
+interface SearchListProps {
+  searchResultsSongs: SpotifyApi.TrackSearchResponse;
+}
+
+export default function SearchList({ searchResultsSongs }: SearchListProps) {
+  const navigate = useNavigate();
+  const { playSelectedSong } = useContext(AppContext);
+
+  function handleOnClick(song: SpotifyApi.TrackObjectFull) {
+    playSelectedSong([song.uri]);
+    navigate("/");
+  }
+
+  return (
+    <List dense sx={{ width: "100%", bgcolor: "background.paper" }}>
+      {searchResultsSongs.tracks.items.map((song) => {
+        const labelId = `checkbox-list-secondary-label-${song}`;
```

Comment:
Have you checked the resulting string? Wondering what `song.toString()` is. Maybe just `[object Object]`.

Reply:
If you want a unqiue string per song use `song.id` just like you did for the React key two lines below.

If you make a value part of a string that is not a string itself (for instance an object), it will automatically be converted to a string by calling its `toString` method. The base implementation for objects just returns "`[object Object]`".

Reply:
I understand what I was doing wrong now. Thanks!